components/password_reset_form/password_reset_form.tsx

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useRef, memo} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {ServerError} from 'mattermost-redux/types/errors';
+import {browserHistory} from 'utils/browser_history';
+import Constants from 'utils/constants';
+import LocalizedInput from 'components/localized_input/localized_input';
+
+import {t} from 'utils/i18n.jsx';
+
+interface Props {
+    location: {search: string };
+    actions: {
+        resetUserPassword: (token: string, newPassword: string) => Promise<{data: any; error: ServerError}>;
+    };
+    siteName?: string;
+}
+
+const PasswordResetForm = ({location, siteName, actions}: Props) => {
+    const [error, setError] = useState<React.ReactNode>(null);
+
+    const passwordInput = useRef<HTMLInputElement>(null);
+
+    const handlePasswordReset = async (e: React.FormEvent) => {
+        e.preventDefault();
+
+        const password = passwordInput.current!.value;
+        if (!password || password.length < Constants.MIN_PASSWORD_LENGTH) {
+            setError(
+                <FormattedMessage
+                    id='password_form.error'
+                    defaultMessage='Please enter at least {chars} characters.'
+                    values={{
+                        chars: Constants.MIN_PASSWORD_LENGTH,
+                    }}
+                />,
+            );
+            return;
+        }
+
+        // setState({error: null});
+
+        const token = (new URLSearchParams(location.search)).get('token');
+
+        if (typeof token !== 'string') {
+            throw new Error('token must be a string');
+        }
+        const {data, error} = await actions.resetUserPassword(token, password);
+        if (data) {
+            browserHistory.push('/login?extra=' + Constants.PASSWORD_CHANGE);
+            setError(null);
+        } else if (error) {
+            setError(error.message);
+        }
+    };
+
+    let errorElement = null;
+    if (error) {
+        errorElement = (
+            <div className='form-group has-error'>
+                <label className='control-label'>
+                    {error}
+                </label>
+            </div>
+        );
+    }
+
+    let formClass = 'form-group';
+    if (error) {
+        formClass += ' has-error';
+    }
```

Comment:
we typically tend to use the `classNames` function to handle conditional classes. See comment below.